dev_setup.sh

```diff
@@ -87,23 +63,160 @@ done
 
 if [ $(id -u) -eq 0 ] && [ "${opt_allowroot}" != true ] ; then
     echo "This script should not be run as root or with sudo."
-    echo "To force, rerun with --allow-root"
+    echo "If you really need to for this, rerun with --allow-root"
     exit 1
 fi
 
-# TODO: Create a setup wizard that guides the user through some decisions
-# if [ ! -f .dev_opts.json ] ; then
-    # E.g.:
-    #  * Run on 'master' or on 'dev'?  Most users probably want 'master'
-    #  * Auto-update?  When on, it will pull and run dev_setup automatically
-    #  * Pull down mimic source?  Most will be happy with just the package
-    #  * Add mycroft-core/bin to the .bashrc PATH?
-
-    # from Picroft's wizard:
-    #   echo '{"use_branch":"master", "auto_update": true}' > .dev_opts.json
-    # or
-    #   echo '{"use_branch":"dev", "auto_update": false}' > .dev_opts.json
-# fi
+CYAN="\e[36m"
+YELLOW="\e[33m"
+RESET="\e[0m"
+HIGHLIGHT=${YELLOW}
+
+# Run a setup wizard the very first time that guides the user through some decisions
+if [ ! -f .dev_opts.json ] ; then
+    sleep 0.5
+    echo
+    echo -e "${CYAN}                    Welcome to Mycroft!  ${RESET}"
+    echo
+    sleep 0.5
+    echo "This script is designed to make working with Mycroft easy.  During this"
+    echo "first run of dev_setup we will ask you a few questions to help setup"
+    echo "your environment."
+    echo
+    sleep 0.5
+    echo "Do you want to run on 'master' or on the 'dev' branch?  Unless you are"
+    echo "a developer modifying mycroft-core itself, you should run on the"
+    echo "'master' branch.  It is updated bi-weekly with a stable release."
+    echo "  Y)es, run on the 'master' branch"
+    echo "  N)o, I want to run against the unstable 'dev' branch"
+    echo -e -n "Choice [${CYAN}Y${RESET}/${CYAN}N${RESET}]: "
```

Comment:
I switched to using multi line strings.  I'm not sure if I really like that style, but maybe I'm  just biased because of the highlighting limitations of nano vs your superior vim highlighting ;)